packaging_automation/package_cloud_statistics_collector.py

```diff
@@ -81,48 +106,101 @@ def fetch_and_save_package_cloud_stats(db_params: DbParams, package_cloud_api_to
-                                                     package_name=package_info['name'],
-                                                     package_version=package_info['version'],
-                                                     package_release=package_info['release'],
-                                                     package_type=package_info['type'],
-                                                     download_date=download_date,
-                                                     download_count=download_count)
-                session.add(pc_stats)
-
-
-def package_historic_statistics_request_address(package_cloud_api_token: str, detail_query_uri: str):
+                    not stat_records_exists(download_date, package_info['filename'],
+                                            package_info['distro_version'], session)):
+                download_detail_record = PackageCloudDownloadDetails(fetch_date=datetime.now(), repo=repo_name,
+                                                                     package_full_name=package_info['filename'],
+                                                                     package_name=package_info['name'],
+                                                                     distro_version=package_info['distro_version'],
+                                                                     package_version=package_info['version'],
+                                                                     package_release=package_info['release'],
+                                                                     package_type=package_info['type'],
+                                                                     epoch=package_info['epoch'],
+                                                                     download_date=download_date,
+                                                                     downloaded_at=downloaded_at,
+                                                                     ip_address=download_detail['ip_address'],
+                                                                     user_agent=download_detail['user_agent'],
+                                                                     source=download_detail['source'],
+                                                                     read_token=download_detail['read_token']
+                                                                     )
+                session.add(download_detail_record)
+
+
+def package_statistics_request_address(package_cloud_api_token: str, detail_query_uri: str):
     return f"https://{package_cloud_api_token}:@packagecloud.io/{detail_query_uri}"
 
 
+def package_statistics_detail_request_address(package_cloud_api_token: str, detail_query_uri: str, page_count: int,
+                                              page_number):
+    return f"https://{package_cloud_api_token}:@packagecloud.io/{detail_query_uri}?per_page={page_count}&page={page_number}"
```

Comment:
The name `page_count` is a bit confusing, sounds like number of pages. Maybe change it to `per_page` in this function, since it is what the url wants.

Reply:
Fixed